spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceBeanCreationFailureAnalyzer.java

```diff
@@ -16,26 +16,68 @@
 
 package org.springframework.boot.autoconfigure.jdbc;
 
+import java.util.Objects;
+
 import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties.DataSourceBeanCreationException;
 import org.springframework.boot.diagnostics.AbstractFailureAnalyzer;
 import org.springframework.boot.diagnostics.FailureAnalysis;
+import org.springframework.core.env.Environment;
+import org.springframework.util.ObjectUtils;
+import org.springframework.util.StringUtils;
 
 /**
  * An {@link AbstractFailureAnalyzer} for failures caused by a
  * {@link DataSourceBeanCreationException}.
  *
  * @author Andy Wilkinson
+ * @author Patryk Kostrzewa
  */
 class DataSourceBeanCreationFailureAnalyzer
 		extends AbstractFailureAnalyzer<DataSourceBeanCreationException> {
 
 	@Override
 	protected FailureAnalysis analyze(Throwable rootFailure,
 			DataSourceBeanCreationException cause) {
-		String message = cause.getMessage();
-		String description = message.substring(0, message.indexOf('.')).trim();
-		String action = message.substring(message.indexOf('.') + 1).trim();
-		return new FailureAnalysis(description, action, cause);
+		return getFailureAnalysis(createDescription(cause), cause);
 	}
 
+	private String createDescription(DataSourceBeanCreationException cause) {
```

Comment:
I don't think that's what I had in mind. What we need to do here is check if "NONE" is active and change the description accordingly.

Reply:
You mean that such detailed description should be provided only in case when connection stands for "NONE"?

Reply:
No that's not what I meant. I think we need to take a step back. Can you  rework your PR to change the exception to expose the `EmbeddedDatabaseConnection`? Then you can adapt the message when it is NONE. The change to fix that issue must  not be that large. Let me know if that works for you.

Reply:
It's ok for me. Actually I've reworked it similarly .